tools/@aws-cdk/prlint/lint.ts

```diff
@@ -16,28 +13,144 @@ class LinterError extends Error {
   }
 }
 
-function createGitHubClient() {
-  const token = process.env.GITHUB_TOKEN;
+interface Test {
+  test: (pr: any, files: any[]) => any;
+}
+
+class ValidateResult {
+  constructor(public errors: string[]) {
+    this.errors = errors;
+  }
 
-  if (token) {
-    console.log("Creating authenticated GitHub Client");
-  } else {
-    console.log("Creating un-authenticated GitHub Client");
+  public isValid() {
+    return this.errors.length === 0;
   }
+}
+
+interface ValidationCollectorOptions {
+  exemption?: (pr: any) => boolean;
+  exemptionMessage?: string;
+  testRules: Test[];
+}
+
+class ValidationCollector {
+  private errors: string[] = [];
```

Comment:
What if this was a map of test name to result. Then we could report the results of all the tests, not just the failed ones. Might be useful to see which ones were successful and which ones were not run.

Reply:
Cool, makes sense 👍